native/src/main/java/io/ballerina/stdlib/sql/parameterprocessor/DefaultResultParameterProcessor.java

```diff
@@ -663,7 +664,8 @@ public Object convertTimestampWithTimezone(java.time.OffsetDateTime offsetDateTi
                 case TypeTags.INT_TAG:
                     return Timestamp.valueOf(offsetDateTime.toLocalDateTime()).getTime();
                 case TypeTags.INTERSECTION_TAG:
-                    return Utils.createTimeStruct(Timestamp.valueOf(offsetDateTime.toLocalDateTime()).getTime());
```

Comment:
the toLocalDateTime() method simply drops the timezone information and returns the date and time.

e.g. 
`2021-07-21 17:14:15 +1:30`   ----> `2021-07-21 17:14:15`